### PR TITLE
fix: [split-view]: update label attribute to be customizable

### PR DESCRIPTION
**Pull request by sarahszhou**
## Description
Bug: https://github.com/adobe/spectrum-web-components/issues/3811
Adjusting the logic such that the label allows for a custom label, if it is defined

I have updated the label constant to consider this.label if it is defined. To test this out, I gave label the value of Resizable Split View Custom Label and ran on Storybook to verify that if label is defined, it overrides the default. See video:

https://github.com/adobe/spectrum-web-components/assets/55859693/e60be647-299a-4fa4-9838-0911eaa580b0

## Related issue(s)

-

## How has this been tested?

-   [x] Storybook 
    1. Give label a string value (manually)
    2. Launch storybook 
    3. Verify through the debugger console that the label is set to the custom label
-   [x] Unit tests
    1. label gets set when specified on split-view

## Types of changes

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)
-   [ ] Chore (minor updates related to the tooling or maintenance of the repository, does not impact compiled assets)

## Checklist

-   [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
-   [ ] My code follows the code style of this project.
-   [ ] If my change required a change to the documentation, I have updated the documentation in this pull request.
-   [ ] I have read the **[CONTRIBUTING](<(https://github.com/adobe/spectrum-web-components/blob/main/CONTRIBUTING.md)>)** document.
-   [ ] I have added tests to cover my changes.
-   [ ] All new and existing tests passed.
-   [ ] I have reviewed at the Accessibility Practices for this feature, see: [Aria Practices](https://www.w3.org/TR/wai-aria-practices/)

## Best practices

This repository uses conventional commit syntax for each commit message; note that the GitHub UI does not use this by default so be cautious when accepting suggested changes. Avoid the "Update branch" button on the pull request and opt instead for rebasing your branch against `main`.

### CODE REVIEW

File: packages/split-view/test/split-view.test.ts [hunk 1]
```diff
@@ -972,6 +972,24 @@ describe('SplitView', () => {
         expect(testPanel).to.be.null;
     });
 
+    it('allows a custom label when resizable if specified', async () => {
```
Comment (susmithayenugula) [inline packages/split-view/test/split-view.test.ts:975]: is there an existing test that falls back to the default string if custom label is not specified? If not, can you add one please?

**Review by audknu — APPROVED** [review]
LGTM 👍 makes sense to only set the label when the split view component can be focused on to resize

File: packages/split-view/test/split-view.test.ts (hunk 1 above)
Comment (sarahszhou) [reply packages/split-view/test/split-view.test.ts:975]: Added one!

**Review by susmithayenugula — APPROVED** [review]

**Comment by sarahszhou** [comment]
Follow up PR for merging to main is on this branch: https://github.com/adobe/spectrum-web-components/pull/4155

**Pull request merged**